dashboard/dashboard.py

```diff
@@ -0,0 +1,777 @@
+            SELECT COUNT(cc.court_case_id), v.verdict, c.court_name
+            FROM court_case as cc
+            JOIN verdict as v ON v.verdict_id = cc.verdict_id
+            JOIN court as c ON c.court_id = cc.court_id
+            GROUP BY v.verdict, c.court_name;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query)
+        result = curs.fetchall()
+    return pd.DataFrame(result)
+
+
+def get_court_data_tags(cnx: connection):
+    """
+    Retrieves court_name data for tags
+    """
+    query = """
+            SELECT COUNT(cc.court_case_id), t.tag_name, c.court_name
+            FROM court_case as cc
+            JOIN tag_assignment as ta ON ta.court_case_id = cc.court_case_id
+            JOIN tag as t ON t.tag_id = ta.tag_id
+            JOIN court as c ON c.court_id = cc.court_id
+            GROUP BY t.tag_name, c.court_name;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query)
+        result = curs.fetchall()
+    return pd.DataFrame(result)
+
+
+def get_court_data_judges(cnx: connection):
+    """
+    Retrieves court_name data for judges
+    """
+    query = """
+            SELECT COUNT(cc.court_case_id), j.judge_name, c.court_name
+            FROM court_case as cc
+            JOIN judge_assignment as ja ON ja.court_case_id = cc.court_case_id
+            JOIN judge as j ON j.judge_id = ja.judge_id
+            JOIN court as c ON c.court_id = cc.court_id
+            GROUP BY j.judge_name, c.court_name;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query)
+        result = curs.fetchall()
+    return pd.DataFrame(result)
+
+
+def get_tag_data_verdict(cnx: connection):
+    """
+    Retrieves tag data for verdicts
+    """
+    query = """
+            SELECT COUNT(cc.court_case_id), v.verdict, t.tag_name
+            FROM court_case as cc
+            JOIN verdict as v ON v.verdict_id = cc.verdict_id
+            JOIN tag_assignment as ta ON ta.court_case_id = cc.court_case_id
+            JOIN tag as t ON t.tag_id = ta.tag_id
+            GROUP BY v.verdict, t.tag_name;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query)
+        result = curs.fetchall()
+    return pd.DataFrame(result)
+
+
+def get_tag_data_judges(cnx: connection):
+    """
+    Retrieves tag data for judges
+    """
+    query = """
+            SELECT COUNT(cc.court_case_id), j.judge_name, t.tag_name
+            FROM court_case as cc
+            JOIN judge_assignment as ja ON ja.court_case_id = cc.court_case_id
+            JOIN judge as j ON j.judge_id = ja.judge_id
+            JOIN tag_assignment as ta ON ta.court_case_id = cc.court_case_id
+            JOIN tag as t ON t.tag_id = ta.tag_id
+            GROUP BY j.judge_name, t.tag_name;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query)
+        result = curs.fetchall()
```

Comment:
This is DRY on steroids, could replace these 3 lines every time with one function.

Reply:
This is a good idea.